content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,201 @@
+
+Surprise! `istio-proxy` is now an `initContainer`.
+
+More specifically, it is an `initContainer` with `restartPolicy: Always` set (a new field, enabled by the `SidecarContainers` feature).
+This tells Kubernetes to treat it as a sidecar.
+
+This means that later containers in the list of `initContainers`, and all normal `containers` will not start until proxy container is first ready.
+Additionally, the pod will terminate even if the proxy container is still running.
+
+### Init container traffic
+
+To put this to the test, lets make our pod actually do something.
+Here we deploy a simple pod that sends a request in an `initContainer`.
+Normally, this would fail.
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sleep
+spec:
+  initContainers:
+  - name: check-traffic
+    image: istio/base
+    command:
+    - curl
+    - httpbin.org/get
+  containers:
+  - name: sleep
+    image: istio/base
+    command: ["/bin/sleep", "infinity"]
+{{< /text >}}
+
+Checking the proxy container, we can see the request both succeeded and went through the Istio sidecar:
+
+{{< text shell >}}
+$ kubectl logs sleep -c istio-proxy | tail -n1
+[2023-07-25T22:00:45.703Z] "GET /get HTTP/1.1" 200 - via_upstream - "-" 0 1193 334 334 "-" "curl/7.81.0" "1854226d-41ec-445c-b542-9e43861b5331" "httpbin.org" ...
+{{< /text >}}
+
+### Exiting pods
+
+Earlier we mentioned that when applications exit (common in `Jobs`), the Pod would live forever.
+Fortunately, this is addressed as well!
+
+First we deploy a pod that will exit after one second and doesn't restart:
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sleep
+spec:
+  restartPolicy: Never
+  containers:
+- name: sleep
+  image: istio/base
+  command: ["/bin/sleep", "1"]
+{{< /text >}}
+
+And we can watch its progress:
+
+{{< text shell >}}
+$ kubectl get pods -w
+NAME    READY   STATUS     RESTARTS   AGE
+sleep   0/2     Init:1/2   0          2s
+sleep   0/2     PodInitializing   0          2s
+sleep   1/2     PodInitializing   0          3s
+sleep   2/2     Running           0          4s
+sleep   1/2     Completed         0          5s
+sleep   0/2     Completed         0          12s
+{{< /text >}}
+
+Here we can see the application container exited, and shortly after Istio's sidecar container exits as well.
+No more zombie pods!
+
+## What about ambient mode?
+
+Last year, Istio announced [ambient mode](/blog/2022/introducing-ambient-mesh/) - a new data plane mode for Istio that doesn't rely on sidecar containers.
+So with that coming, does any of this even matter?
+
+I would say a resounding "Yes"!
+
+While the impacts of sidecar are lessened when ambient mode is used for a workload, I expect that almost all large scale Kubernetes users have some sort of sidecar in their deployments.
+This could be Istio workloads they don't want to migrate to ambient, that they haven't *yet* migrated, or things unrelated to Istio.
```

Comment:
can it be Istio with the very "special" configuration of an envoy that ambient cannot provide? I don't know enough to suggest any specifics, but from monitoring background - whenever we say ambient or central, all customizations are being lost immediately.

Reply:
Yeah that is one of the primary cases we expect them to be kept (all theoretical, since we haven't properly launched it yet to get real feedback!)